File: moedict-android/app/src/main/java/com/yichiuan/moedict/data/DataModule.java
package com.yichiuan.moedict.data;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;


@Module
public class DataModule {
    @Provides
    @Singleton
    MoeRepository provideMoeRepository(Context context) {
        return new MoeRepository(context);
    }
}

File: moedict-android/app/src/main/java/com/yichiuan/moedict/data/DataModule.kt
package com.yichiuan.moedict.data

import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton


@Module
class DataModule {
    @Provides
    @Singleton
    fun provideMoeRepository(context: Context): MoeRepository {
        return MoeRepository(context)
    }
}
